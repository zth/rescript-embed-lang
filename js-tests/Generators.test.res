open TestFramework
open RescriptEmbedLang__Internal

module Fs = NodeJs.Fs
module Path = NodeJs.Path

describe("findContentInFile", () => {
  let testFile = Path.resolve([NodeJs.Process.process->NodeJs.Process.cwd, "js-tests/TestFile.txt"])
  let testFile2 = Path.resolve([
    NodeJs.Process.process->NodeJs.Process.cwd,
    "js-tests/TestFile2.txt",
  ])

  testAsync("finds content", async () => {
    let foundContent = await testFile->findContentInFile(["%edgeql", "%css"])
    expect(foundContent)->Expect.toMatchSnapshot
  })

  testAsync("finds content 2", async () => {
    let foundContent = await testFile2->findContentInFile(["%edgeql"])
    Console.log(foundContent)
    expect(foundContent)->Expect.toMatchSnapshot
  })
})
