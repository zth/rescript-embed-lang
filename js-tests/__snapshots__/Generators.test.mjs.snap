// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`findContentInFile finds content 1`] = `
[
  {
    "content": [
      "",
      "  # @name TestQuery",
      "  select User {",
      "    name",
      "  } filter .name = 'test'",
      "",
    ],
    "end": {
      "col": 0,
      "line": 5,
    },
    "start": {
      "col": 21,
      "line": 0,
    },
    "tag": "%edgeql",
  },
  {
    "content": [
      "",
      "  display: flex;",
      "",
    ],
    "end": {
      "col": 0,
      "line": 18,
    },
    "start": {
      "col": 20,
      "line": 16,
    },
    "tag": "%css",
  },
]
`;

exports[`findContentInFile finds content 2 1`] = `
[
  {
    "content": [
      "",
      "    # @name allMovies",
      "    select Movie {",
      "      id,",
      "      title",
      "      actors: {",
      "          id,",
      "          name,",
      "          numberOfPets := count(.pets)",
      "      }",
      "    } order by .title",
      "    ",
    ],
    "end": {
      "col": 4,
      "line": 12,
    },
    "start": {
      "col": 23,
      "line": 1,
    },
    "tag": "%edgeql",
  },
  {
    "content": [
      "",
      "    # @name allMoviesNested",
      "    select Movie {",
      "      id,",
      "      title,",
      "      actors: {",
      "          id,",
      "          name,",
      "          numberOfPets := count(.pets)",
      "      }",
      "    } order by .title",
      "    ",
    ],
    "end": {
      "col": 4,
      "line": 36,
    },
    "start": {
      "col": 23,
      "line": 25,
    },
    "tag": "%edgeql",
  },
  {
    "content": [
      "",
      "    # @name movieByTitle",
      "    select Movie {",
      "      id,",
      "      title,",
      "      actors: {",
      "          id,",
      "          name,",
      "          numberOfPets := count(.pets),",
      "          typesDump: {",
      "            date,",
      "            localDateTime,",
      "            localDate,",
      "            relativeDuration,",
      "            duration,",
      "            dateDuration,",
      "            localTime,",
      "            json",
      "          }",
      "      }",
      "    } ",
      "      filter .title = <str>$title",
      "      limit 1",
      "    ",
    ],
    "end": {
      "col": 4,
      "line": 63,
    },
    "start": {
      "col": 23,
      "line": 40,
    },
    "tag": "%edgeql",
  },
  {
    "content": [
      "",
      "  # @name AddActor",
      "  insert Person {",
      "    name := <str>$name",
      "  }",
      "",
    ],
    "end": {
      "col": 0,
      "line": 75,
    },
    "start": {
      "col": 27,
      "line": 70,
    },
    "tag": "%edgeql",
  },
  {
    "content": [
      "",
      "  # @name RemoveActor",
      "  delete Person filter .id = <uuid>$id",
      "",
    ],
    "end": {
      "col": 0,
      "line": 82,
    },
    "start": {
      "col": 30,
      "line": 79,
    },
    "tag": "%edgeql",
  },
]
`;

exports[`findContentInFile finds content 3 1`] = `
[
  {
    "content": [
      "",
      "    # @name allMovies",
      "    select Movie {",
      "      id,",
      "      title",
      "      actors: {",
      "          id,",
      "          name,",
      "          numberOfPets := count(.pets)",
      "      }",
      "    } order by .title",
      "    ",
    ],
    "end": {
      "col": 4,
      "line": 12,
    },
    "start": {
      "col": 23,
      "line": 1,
    },
    "tag": "%edgeql",
  },
  {
    "content": [
      "",
      "    # @name allMoviesNested",
      "    select Movie {",
      "      id,",
      "      title,",
      "      actors: {",
      "          id,",
      "          name,",
      "          numberOfPets := count(.pets)",
      "      }",
      "    } order by .title",
      "    ",
    ],
    "end": {
      "col": 4,
      "line": 36,
    },
    "start": {
      "col": 23,
      "line": 25,
    },
    "tag": "%edgeql",
  },
  {
    "content": [
      "",
      "    # @name movieByTitle",
      "    select Movie {",
      "      id,",
      "      title,",
      "      actors: {",
      "          id,",
      "          name,",
      "          numberOfPets := count(.pets),",
      "          typesDump: {",
      "            date,",
      "            localDateTime,",
      "            localDate,",
      "            relativeDuration,",
      "            duration,",
      "            dateDuration,",
      "            localTime,",
      "            json",
      "          }",
      "      }",
      "    } ",
      "      filter .title = <str>$title",
      "      limit 1",
      "    ",
    ],
    "end": {
      "col": 4,
      "line": 63,
    },
    "start": {
      "col": 23,
      "line": 40,
    },
    "tag": "%edgeql",
  },
  {
    "content": [
      "",
      "  # @name AddActor",
      "  insert Person {",
      "    name := <str>$name",
      "  }",
      "",
    ],
    "end": {
      "col": 0,
      "line": 75,
    },
    "start": {
      "col": 27,
      "line": 70,
    },
    "tag": "%edgeql",
  },
  {
    "content": [
      "",
      "  # @name RemoveActor",
      "  delete Person filter .id = <uuid>$id",
      "",
    ],
    "end": {
      "col": 0,
      "line": 82,
    },
    "start": {
      "col": 30,
      "line": 79,
    },
    "tag": "%edgeql",
  },
]
`;
